src/fides/api/models/privacy_experience.py

```diff
@@ -1,94 +1,252 @@
+    """
+
+    def generate_uuid(self) -> str:
+        """
+        Generates a uuid with a prefix based on the tablename to be used as the
+        record's ID value
+        """
+        try:
+            prefix = f"{self.current_column.table.name[:3]}_"  # type: ignore
+        except AttributeError:
+            prefix = ""
+        uuid = str(uuid4())
+        return f"{prefix}{uuid}"
+
+    # Overrides Base.id so this is not a primary key.
+    # Instead, we have a composite PK of notice_id and experience_config_id
+    id = Column(String(255), default=generate_uuid)
+
+    notice_id = Column(
+        String,
+        ForeignKey("privacynotice.id", ondelete="CASCADE"),
+        index=True,
+        nullable=False,
+        primary_key=True,
+    )
+    experience_config_id = Column(
+        String,
+        ForeignKey("privacyexperienceconfig.id", ondelete="CASCADE"),
+        index=True,
+        nullable=False,
+        primary_key=True,
+    )
 
 
 class ComponentType(Enum):
     """
     The component type - not formalized in the db
+
+    Overlay type has been deprecated but can't be removed for backwards compatibility
+    without significant data migrations.
     """
 
-    overlay = "overlay"
+    overlay = "overlay"  # Deprecated. DO NOT REMOVE.
+    banner_and_modal = "banner_and_modal"
+    modal = "modal"
     privacy_center = "privacy_center"
-    tcf_overlay = "tcf_overlay"
+    tcf_overlay = "tcf_overlay"  # TCF Banner + modal combined
+
+
+# Fides JS UX Types - there should only be one of these defined per region
+FidesJSUXTypes: List[ComponentType] = [
+    ComponentType.banner_and_modal,
+    ComponentType.modal,
+]
 
 
-class BannerEnabled(Enum):
+class PrivacyExperienceConfigBase:
     """
-    Whether the banner should display - not formalized in the db
+    Common fields shared between:
+    - ExperienceConfigTemplate
+    - PrivacyExperienceConfig
+    - PrivacyExperienceConfigHistory
+
+    These are non-translated fields.
     """
 
-    always_enabled = "always_enabled"
-    enabled_where_required = "enabled_where_required"  # If the user's region has at least one opt-in or notice-only notice
-    always_disabled = "always_disabled"
+    allow_language_selection = Column(
+        Boolean, nullable=False, default=True, server_default="t"
+    )
+
+    @declared_attr
+    def component(cls) -> Column:
+        return Column(
+            EnumColumn(ComponentType),
+            nullable=False,
+            index=True,
+        )
+
+    disabled = Column(Boolean, nullable=False, default=True)
+    dismissable = Column(Boolean, nullable=False, default=True, server_default="t")
+    name = Column(String)
+
+
+class ExperienceConfigTemplate(PrivacyExperienceConfigBase, Base):
+    """Table for out-of-the-box Experience Configurations"""
+
+    privacy_notice_keys = Column(
+        ARRAY(String)
+    )  # A list of notice keys which should correspond to a subset of out-of-the-box notice keys
```

Comment:
My current draft of creating Experience Config out of the box templates specifies the notices by key in the template, not by id, to make this more human readable.  However, when creating Privacy Experience Configs directly, the API request should specify notice ids.  There's also a side constraint right now that out of the box Notices need to have unique keys.